### PR TITLE
Support low rank adaptation for all matrices in attention blocks

### PR DESCRIPTION
**Pull request by caroteu**
@anwai98 
Adding adapters to the key matrices and feed forward layers does not add any significant overload in the memory while increasing the number of trainable parameters. For example if we add late lora to blocks 6-11:

Only on Q, V: 10268MiB, ~15.68M Params
On Q, V, K, MLP: 10296MiB, ~17.45M Params

### CODE REVIEW

File: micro_sam/models/peft_sam.py [hunk 1]
```diff
@@ -365,6 +365,20 @@ def __init__(self, block: nn.Module):
...
+
+    def __init__(self, block: nn.Module):
+        super().__init__(block=block)
+        self.block = block
+
+        for k, v in self.block.named_parameters():
+            v.requires_grad = True
+
+    def forward(self, x):
+        return x
```
Comment (anwai98) [inline micro_sam/models/peft_sam.py:379]: Just to be sure @caroteu, we need this so that we allow gradient updates on all layers in a block and not just the attention matrices, right?

Comment (anwai98) [reply micro_sam/models/peft_sam.py:379]: I see. So in theory, we desire to "unfreeze" all parts of the attention block?

Comment (anwai98) [reply micro_sam/models/peft_sam.py:379]: Nice. Thanks for clarifying!

File: finetuning/livecell_finetuning.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ def finetune_livecell(args):
     train_loader, val_loader = get_dataloaders(patch_shape=patch_shape, data_path=args.input_path)
     scheduler_kwargs = {"mode": "min", "factor": 0.9, "patience": 10, "verbose": True}
 
+    # freeze encoder: 35.02 GB
+    # late lora 50% classic: 43.142 GB
+    # late FT 50%: 42.87 GB
+    # lora classic: 48.46 GB
+    # full ft: 49.35
+
```
Comment (anwai98) [inline finetuning/livecell_finetuning.py:64]: I can reproduce the memory efficiency on LIVECell atleast. I am very confident now that there is some stochastic memory issues you are facing on HPA.

cc: @caroteu

**Comment by anwai98** [comment]
I think this is in a good state. I'll go ahead and merge this in the running Late PEFT main PR, once the tests pass!

cc: @caroteu
